lib/galaxy/tools/data/__init__.py

```diff
@@ -896,11 +899,338 @@ def _remove_entry(self, values):
         raise NotImplementedError("Not supported")
 
 
+class DatabaseToolDataTable(ToolDataTable, Dictifiable):
+    """
+    Data stored in the database, mainly for highly transactional data::
+
+        <table type="database" name="test">
+            <column name='...' index = '...' />
+        </table>
+
+    """
+    dict_collection_visible_keys = ['name']
+    type_key = 'database'
+
+    datatable_name = None
+    datatable_allow_duplicate_entries = True
+    sa_session = None
+    datatable_id = None
+
+    def __init__(self, config_element, tool_data_path, from_shed_config=False, filename=None, tool_data_path_files=None, other_config_dict=None, app=None):
```

Comment:
Can you get by without passing ``app`` all the way through here. Maybe just pass ``app.data_tables_model`` (if available)? I think that is all you land up using (that and the tool_shed name link - which you can just assume you cannot load these if you don't have the model).

Reply:
Thanks for the suggestion, I've now limited it to just the data_tables_model!